sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala

```diff
@@ -54,10 +54,6 @@ object ResolveHints {
 
       val newNode = CurrentOrigin.withOrigin(plan.origin) {
         plan match {
-          case r: UnresolvedRelation =>
```

Comment:
actually you can't remove this case entirely. you still need to match on UnresolvedRelation, but just use `r.tableIdentifier.table`